docs/source/reference/core.rst

```diff
@@ -0,0 +1,30 @@
+.. module:: optuna
+
+Core
+====
+
+.. autofunction:: create_study
+
+.. seealso::
+    :func:`optuna.study.create_study`.
+
+.. autofunction:: load_study
+
+.. seealso::
+    :func:`optuna.study.load_study`.
+
+.. autofunction:: delete_study
+
+.. seealso::
+    :func:`optuna.study.delete_study`.
+
+.. autofunction:: get_all_study_summaries
+
+.. seealso::
+    :func:`optuna.study.get_all_study_summaries`.
+
+.. autoclass:: TrialPruned
+    :members:
+
+.. seealso::
+    :class:`optuna.exceptions.TrialPruned`.
```

Comment:
Not a strong opinion but how about clarifying the relationship, that "This is an alias for :class:`optuna.exceptions.TrialPruned`" since the purpose isn't obvious otherwise?